{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "MEMBER",
    "body": "**Description of the problem**\r\n\r\nIn recent runs (https://github.com/GenericMappingTools/pygmt/actions/runs/560287188), `test_load_libgmt_with_a_bad_library_path` fails with GMT dev on Linux and macOS, but pass on Windows, possibly because we merged #702. However, it passed when @weiji14 triggered the dev tests before merging (https://github.com/GenericMappingTools/pygmt/actions/runs/560286224).\r\n\r\nIt's unclear to me what happens. We can wait for today's scheduled runs and see if it works or not.\r\n\r\nSee the error messages below:\r\n```\r\n___________________ test_load_libgmt_with_a_bad_library_path ___________________\r\n\r\nmonkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f462205dca0>\r\n\r\n    def test_load_libgmt_with_a_bad_library_path(monkeypatch):\r\n        \"\"\"\r\n        Test that loading still works when given a bad library path.\r\n        \"\"\"\r\n        # Set a fake \"GMT_LIBRARY_PATH\"\r\n        monkeypatch.setenv(\"GMT_LIBRARY_PATH\", \"/not/a/real/path\")\r\n>       assert check_libgmt(load_libgmt()) is None\r\n\r\n/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pygmt/tests/test_clib_loading.py:48: \r\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \r\n\r\n    def load_libgmt():\r\n        \"\"\"\r\n        Find and load ``libgmt`` as a :py:class:`ctypes.CDLL`.\r\n    \r\n        By default, will look for the shared library in the directory specified by\r\n        the environment variable ``GMT_LIBRARY_PATH``. If it's not set, will let\r\n        ctypes try to find the library.\r\n    \r\n        Returns\r\n        -------\r\n        :py:class:`ctypes.CDLL` object\r\n            The loaded shared library.\r\n    \r\n        Raises\r\n        ------\r\n        GMTCLibNotFoundError\r\n            If there was any problem loading the library (couldn't find it or\r\n            couldn't access the functions).\r\n        \"\"\"\r\n        lib_fullnames = []\r\n        error = True\r\n        for libname in clib_full_names():\r\n            lib_fullnames.append(libname)\r\n            try:\r\n                libgmt = ctypes.CDLL(libname)\r\n                check_libgmt(libgmt)\r\n                error = False\r\n                break\r\n            except OSError as err:\r\n                error = err\r\n        if error:\r\n>           raise GMTCLibNotFoundError(\r\n                \"Error loading the GMT shared library \"\r\n                f\"{', '.join(lib_fullnames)}.\\n {error}.\"\r\n            )\r\nE           pygmt.exceptions.GMTCLibNotFoundError: Error loading the GMT shared library libgmt.so.\r\nE            libgmt.so: cannot open shared object file: No such file or directory.\r\n\r\n```",
    "closed_at": "2021-02-12T22:58:55Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
        "events_url": "https://api.github.com/users/seisman/events{/privacy}",
        "followers_url": "https://api.github.com/users/seisman/followers",
        "following_url": "https://api.github.com/users/seisman/following{/other_user}",
        "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/seisman",
        "id": 3974108,
        "login": "seisman",
        "node_id": "MDQ6VXNlcjM5NzQxMDg=",
        "organizations_url": "https://api.github.com/users/seisman/orgs",
        "received_events_url": "https://api.github.com/users/seisman/received_events",
        "repos_url": "https://api.github.com/users/seisman/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/seisman"
    },
    "comments": 5,
    "comments_content": [
        {
            "author_association": "MEMBER",
            "body": "Here is what happens. For GMT dev builds,\r\n\r\n**Windows:** We use the conda-forge's GMT dev pacakges because we can't (don't) build GMT source codes on Windows. Obviously, conda adds the bin path to PATH so that we can call `gmt --show-library` to find the library, and the test passes.\r\n\r\n**Linux/macOS**: We build GMT source codes and install it to `${{ github.workspace }}/gmt-install-dir`, and we set `GMT_LIBRARY_PATH` to `${{ github.workspace }}/gmt-install-dir/lib` but don't add it's bin path to PATH, so PyGMT can't find the GMT library.\r\n\r\nWe have two ways to fix the problem:\r\n\r\n1. On Linux/macOS, add GMT's bin path to PATH\r\n2. Remove the test, because we're going to add more comprehensive tests in #872",
            "created_at": "2021-02-12T19:34:52Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874#issuecomment-778407909",
            "id": 778407909,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc3ODQwNzkwOQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778407909/reactions"
            },
            "updated_at": "2021-02-12T19:39:28Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778407909",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> * Remove the test, because we're going to add more comprehensive tests in #872\r\n\r\nMaybe no, because the new tests in #872 need GMT's bin in PATH.\r\n\r\n> * On Linux/macOS, add GMT's bin path to PATH\r\n\r\nThis is the preferred way.\r\n\r\n\r\nThoughts?",
            "created_at": "2021-02-12T19:48:40Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874#issuecomment-778415838",
            "id": 778415838,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc3ODQxNTgzOA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778415838/reactions"
            },
            "updated_at": "2021-02-12T19:48:40Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778415838",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "> **Linux/macOS**: We build GMT source codes and install it to `${{ github.workspace }}/gmt-install-dir`, and we set `GMT_LIBRARY_PATH` to `${{ github.workspace }}/gmt-install-dir/lib` but don't add it's bin path to PATH, so PyGMT can't find the GMT library.\r\n\r\nI don't get it, so how did `pygmt/clib/loading.py` find the GMT bin before #702 even without us setting it? This seems like a regression bug to me.",
            "created_at": "2021-02-12T22:00:18Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874#issuecomment-778478345",
            "id": 778478345,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc3ODQ3ODM0NQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778478345/reactions"
            },
            "updated_at": "2021-02-12T22:00:18Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778478345",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "The test is new in #702, no?",
            "created_at": "2021-02-12T22:25:07Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874#issuecomment-778487896",
            "id": 778487896,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc3ODQ4Nzg5Ng==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778487896/reactions"
            },
            "updated_at": "2021-02-12T22:25:07Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778487896",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
                "events_url": "https://api.github.com/users/seisman/events{/privacy}",
                "followers_url": "https://api.github.com/users/seisman/followers",
                "following_url": "https://api.github.com/users/seisman/following{/other_user}",
                "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/seisman",
                "id": 3974108,
                "login": "seisman",
                "node_id": "MDQ6VXNlcjM5NzQxMDg=",
                "organizations_url": "https://api.github.com/users/seisman/orgs",
                "received_events_url": "https://api.github.com/users/seisman/received_events",
                "repos_url": "https://api.github.com/users/seisman/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/seisman"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Well, that test (`test_load_libgmt_fails`) was actually modified from a previous test. Before, the library loading would just fail when a fake \"GMT_LIBRARY_PATH\" is set because we didn't have extra options. After, the library loading should pass because we check for other paths (using `gmt --show-library` and standard system paths). I think you're right that we still need to set GMT's bin in the system PATH.",
            "created_at": "2021-02-12T22:54:55Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874#issuecomment-778500434",
            "id": 778500434,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
            "node_id": "MDEyOklzc3VlQ29tbWVudDc3ODUwMDQzNA==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778500434/reactions"
            },
            "updated_at": "2021-02-12T22:54:55Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/778500434",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874/comments",
    "created_at": "2021-02-12T19:19:29Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/874",
    "id": 807485890,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWU4MDc0ODU4OTA=",
    "number": 874,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874/timeline",
    "title": "test_load_libgmt_with_a_bad_library_path fails with GMT dev versions on Linux and macOS",
    "updated_at": "2021-02-12T22:58:55Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/874",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
        "events_url": "https://api.github.com/users/seisman/events{/privacy}",
        "followers_url": "https://api.github.com/users/seisman/followers",
        "following_url": "https://api.github.com/users/seisman/following{/other_user}",
        "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/seisman",
        "id": 3974108,
        "login": "seisman",
        "node_id": "MDQ6VXNlcjM5NzQxMDg=",
        "organizations_url": "https://api.github.com/users/seisman/orgs",
        "received_events_url": "https://api.github.com/users/seisman/received_events",
        "repos_url": "https://api.github.com/users/seisman/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/seisman"
    }
}